engine/engine_test.go

```diff
@@ -112,6 +113,63 @@ func TestQueryExplain(t *testing.T) {
 	}
 }
 
+func TestQueryAnalyze(t *testing.T) {
+	opts := promql.EngineOpts{Timeout: 1 * time.Hour}
+	series := storage.MockSeries(
+		[]int64{240, 270, 300, 600, 630, 660},
+		[]float64{1, 2, 3, 4, 5, 6},
+		[]string{labels.MetricName, "foo"},
+	)
+
+	start := time.Unix(0, 0)
+	end := time.Unix(1000, 0)
+
+	// Calculate concurrencyOperators according to max available CPUs.
+	totalOperators := runtime.GOMAXPROCS(0) / 2
+	concurrencyOperators := []engine.AnalyzeOutputNode{}
+	for i := 0; i < totalOperators; i++ {
+		concurrencyOperators = append(concurrencyOperators, engine.AnalyzeOutputNode{
+			OperatorTelemetry: &model.TimingInformation{}, Children: []engine.AnalyzeOutputNode{
+				{OperatorTelemetry: &model.TimingInformation{}},
+			},
+		})
+	}
+
+	for _, tc := range []struct {
+		query    string
+		expected *engine.AnalyzeOutputNode
```

Comment:
Good stuff but the test could be improved even further if we would check whether the [CPUTime](https://github.com/thanos-io/promql-engine/pull/289/files#diff-a738ddf76b5904ba1562ec7baa7ddd82de3c3303bc8e1d006e24ba990df9a4edR18) has some value. We could add `assertFn func(t *testing.T, got *engine.AnalyzeOutputNode)` here or something like that. Inside there you could write an assertion that `got.CPUTime > 0`.